addons/portal/data/portal_data.xml

```diff
@@ -34,5 +34,14 @@
 <p>
 ${object.wizard_id.welcome_message or ''}]]></field>
         </record>
+        <template id="portal_share_document_template">
+            <div>
+                <p>Hello,</p>
+                <p>A document <strong t-esc="record_name"/> has been shared with you.</p>
```

Comment:
I would directly give record as a recordset to the template rendering. That way it is easier to customize it if necessary (record.display_name will then be used here).